packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/CustomizeColumnsModal.js

```diff
@@ -62,7 +65,8 @@ const CustomizeColumnsModal = ({
   const onCheckboxCheck = (col, value) => {
     const changedDefinitions = columnObjects.map((definition) => {
       if (definition.id === col.id) {
-        return { ...definition, isVisible: value };
+        definition.isVisible = value;
+        return definition;
       }
```

Comment:
The original code should work and is a better approach, because it creates a new object with the new `isVisible` value (as opposed to mutating the `definition` object).

Reply:
Reverted to `return { ...definition, isVisible: value };`

Reply:
Implemented logic that differentiates array parameters and object parameters.